drivers/note/note_lastsched_driver.c

```diff
@@ -0,0 +1,397 @@
+{
+  note_lastsched_common(tcb, NOTE_RESUME, (void *)cpu);
+}
+
+static void note_lastsched_cpu_resumed(FAR struct note_driver_s *drv,
+                                       FAR struct tcb_s *tcb)
+{
+  note_lastsched_common(tcb, NOTE_RESUME, NULL);
+}
+#  endif
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
+static void note_lastsched_premption(FAR struct note_driver_s *drv,
+                                     FAR struct tcb_s *tcb, bool locked)
+{
+  note_lastsched_common(
+    tcb, locked ? NOTE_PREEMPT_LOCK : NOTE_PREEMPT_UNLOCK, NULL);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_CSECTION
+static void note_lastsched_csection(FAR struct note_driver_s *drv,
+                                    FAR struct tcb_s *tcb, bool enter)
+{
+  note_lastsched_common(
+      tcb, enter ? NOTE_CSECTION_ENTER : NOTE_CSECTION_LEAVE, NULL);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SPINLOCKS
+static void note_spinlock(FAR struct note_driver_s *drv,
+                          FAR struct tcb_s *tcb, FAR volatile void *spinlock,
+                          int type)
+{
+  note_lastsched_common(tcb, type, (FAR void *)spinlock);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_IRQHANDLER
+static void note_lastsched_irqhandler(FAR struct note_driver_s *drv, int irq,
+                                      FAR void *handler, bool enter)
+{
+  note_lastsched_common(this_task(), enter ? NOTE_IRQ_ENTER : NOTE_IRQ_LEAVE,
+                        (FAR void *)irq);
+}
+#endif
+
+#ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
+static void note_lastsched_syscall_enter(FAR struct note_driver_s *drv,
+                                         int nr, int argc, va_list *ap)
+{
+  note_lastsched_common(this_task(), NOTE_SYSCALL_ENTER, (void *)nr);
+}
+
+static void note_lastsched_syscall_leave(FAR struct note_driver_s *drv,
+                                         int nr, uintptr_t result)
+{
+  note_lastsched_common(this_task(), NOTE_SYSCALL_LEAVE, (void *)nr);
+}
+#endif
+
+/****************************************************************************
+ * Name: note_lastched_register
+ ****************************************************************************/
+
+int note_lastched_register(void)
+{
+  g_note_lastsched.freq = up_perf_getfreq();
+  if (g_note_lastsched.freq == 0)
+    {
+      g_note_lastsched.freq = NSEC_PER_SEC;
+    }
+
+  clock_gettime(CLOCK_REALTIME, &g_note_lastsched.ts);
+  return note_driver_register(&g_note_lastsched_driver);
+}
+
+/****************************************************************************
+ * Name: note_lastsched_dump
+ ****************************************************************************/
+
+void note_lastsched_dump(void)
+{
+  size_t i;
+  uint32_t freq;
+  irqstate_t flags;
+  struct note_filter_mode_s old;
+  size_t index = g_note_lastsched.index;
+  FAR struct note_lastsched_chunk_s *note;
+
+  sched_note_filter_mode(&old, NULL);
```

Comment:
why need if nuttx is in the panic

Reply:
syslog in notelast is a system call, but we don't care about it.
sched_note needs to be turned off when dumping